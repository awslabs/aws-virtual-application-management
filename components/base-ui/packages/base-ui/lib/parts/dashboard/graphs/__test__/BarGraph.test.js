/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import BarGraph from '../BarGraph';
import { blueDatasets } from '../graph-options';

jest.mock('react-chartjs-2', () => ({ HorizontalBar: () => null }));

describe('BarGraph', () => {
  it('renders correctly', () => {
    const title = 'Tasks';
    const data = {
      labels: ['Eat', 'Run', 'Walk', 'Sleep', 'Work'],
      datasets: blueDatasets(title, [1, 8, 5, 6, 3]),
    };
    const barGraph = renderer.create(<BarGraph className="mr4" data={data} title="Tasks" />).toJSON();
    expect(barGraph).toMatchSnapshot();
  });
});
