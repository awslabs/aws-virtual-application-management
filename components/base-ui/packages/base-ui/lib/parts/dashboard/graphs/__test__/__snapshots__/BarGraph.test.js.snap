// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarGraph renders correctly 1`] = `
<div
  className="mr4"
>
  <div
    className="fs-9 center mt1 mb1"
  >
    Tasks
  </div>
  <div />
</div>
`;
